heron/common/tests/cpp/network/switch_unittest.cpp

```diff
@@ -108,15 +110,20 @@ void terminate_server(sp_uint32 port) {
 }
 
 void start_test(sp_int32 nclients, sp_uint64 requests) {
+  sp_uint32 server_port = 0;
+  CountDownLatch* latch = new CountDownLatch(1);
+
   // start the server thread
-  std::thread sthread(start_server, SERVER_PORT);
+  std::thread sthread(start_server, &server_port, latch);
```

Comment:
The latch solves the race condition.
The server thread updates server_port, then latch is released, then the client thread can read server_port.